### ISSUE TITLE
Experiment: CODEOWNERS for VAOS

### ISSUE THREAD

**Issue by johnpaulashenfelter** [department-of-veterans-affairs/va.gov-team#9581]
## TL;DR
We need to find out if a VFS team can work in isolation with the code specific to their project without manual VSP review.

## Background
One possible way to isolate VFS teams is to let them take control of their own code review without additional manual oversight steps from VSP developers. This potentially allows the team to move faster, addresses an oft-cited pain point of the wait for reviews, and removes a blocker to scaling up with more VFS teams.

## Experimental Design/Methods
We chose an experienced team with a long history on the project in multiple roles. The VAOS team has engineers that have worked on front-end and back-end applications for years with many commits to the `vets-api` codebase. They are experienced in Rails, the VSP platform, and the current VSP process. This team represents our “best case scenario” so we would expect their results to be an upper bound on success.

This team also represents the lowest possible risk since they are experienced in the tools,  platform, and process. They have been through many VSP code reviews with very few change requests (1).  They have also at times felt like the code review process slowed them down based on personal communication and Sprint Design team feedback.

The CODEOWNERS file, which is used by GitHub to ensure appropriate manual code review, will be adjusted to allow the @department-of-veterans-affairs/vfs-vaos team to review pull requests in their module and associated test support files.

Pull requests that include “common” code, such as anything in `app/` or `config/` directories, will go through the existing manual code review process.

Automated review using Rubocop, Codeclimate, and automate RSpec testing and test coverage statistics will still be applied to the team-specific pull requests with the normal expectations for quality and consistency.

The primary results will compare GitHub pull request data for the previous two sprints with the upcoming two sprints to find qualitative and quantitative differences between the treatments. The team will also be interviewed to understand how they feel the treatment affected their work. We do not expect statistically significant data from this experiment.

## Preliminary Data
GitHub pull request data from the previous two sprints for the VAOS team will be collected and retroactively categorized into VSP- and team-reviewed    groups to provide a rough control for the experiment in this narrow instance.

## Results
TBD

**Comment by johnpaulashenfelter** [comment]
Baseline data in https://docs.google.com/spreadsheets/d/1BSWTDro_tr8JAuuda9NU7Ry5LwBAKPrzwGA-Es2IZ60/edit#gid=1573886863

**Comment by rroueche** [comment]
@alexpappasoddball and @johnpaulashenfelter this is really exciting, thanks for the great write-up here. I'm looking forward to a future readout on how the experiment went, what you hypotheses you validated as true / false, etc.

In your TL;DR and Background here (and in experiment itself!) can you make sure you're considering not only how to enable VFS teams to work on their own and work faster, but how to do that **without putting VA.gov quality at risk**?

**Comment by johnpaulashenfelter** [comment]
For VAOS, both before and after 50+% of the PRs were completely related to code within the module (modules/vaos and corresponding specs) that they are responsible for.
